src/Provider/OGCWebServiceHelper.js

```diff
@@ -23,62 +17,47 @@ const tileCoord = new Extent('WMTS:WGS84G', 0, 0, 0);
 
 export default {
     getColorTextureByUrl(url, networkOptions) {
-        if (cache.has(url)) {
-            return Promise.resolve(cache.get(url));
+        if (Cache.has(url)) {
+            return Promise.resolve(Cache.get(url));
         }
 
-        const promise = (pending.has(url)) ?
-            pending.get(url) :
-            Fetcher.texture(url, networkOptions);
-
-        pending.set(url, promise);
-
-        return promise.then((texture) => {
+        const promise = Fetcher.texture(url, networkOptions).then((texture) => {
             texture.generateMipmaps = false;
             texture.magFilter = THREE.LinearFilter;
             texture.minFilter = THREE.LinearFilter;
             texture.anisotropy = 16;
 
-            if (!cache.has(url)) {
-                cache.set(url, texture);
-            }
-
-            pending.delete(url);
+            Cache.set(url, texture, Cache.POLICIES.TEXTURE);
             return texture;
         });
+
+        Cache.set(url, promise, Cache.POLICIES.PENDING);
```

Comment:
Can't we remove this `PENDING` thing and instead always store a `Promise` in the cache?
```js
if (Cache.has(url)) {
   return Cache.get(url);
}
const promise = Fetcher.texture(...).then((texture) => {
   ...
   return texture;
});
Cache.set(url, promise);
return promise;
```
?

Reply:
I don't think it's a problem since commands are already executed in an async context (see https://github.com/iTowns/itowns/blob/master/src/Core/Scheduler/Scheduler.js#L154).

Reply:
So, if I remove those policies, would that mean dropping the time-based lifetime in favor of something size-based ? If you feel that way too, it may be better to go in the suggestion of @mbredif in #741.

Reply:
I'm not sure how the removal of the `PENDING` policy is related to entirely dropping the time-based lifetime?

IMHO if this PR already unifies the caching mechanism + provides a way to remove outdated entries + simplifies the providers code it's already a big win.

Then we can keep working on this to improve it:
- add support for pre-loading/pre-caching (e.g: the application knows that in a near future it will display a currently hidden pointcloud and wants to start loading it now)
- allow itowns (eg LayeredMaterialNodeProcessing.js) to query the cache and adapt its behavior (no need to run 10s of async commands if it knows that everything needed is already available in the cache)

Reply:
Oh I may have misunderstand the initial remark, sorry about that. It does not impact `TileProvider` at all, so I just updated for what you suggested.